.github/workflows/CI.yml

```diff
@@ -267,12 +267,12 @@ jobs:
         shell: C:\cygwin64\bin\bash.exe --login -o igncr '{0}'
         run: dev/ci.sh
 
-  slack-notification-on-failure:
-    name: Send slack notification on CI failure
+  slack-notification:
+    name: Send Slack notification on status change
     needs:
       - test-unix
       - test-cygwin
-    if: ${{ always() && github.event_name == 'push' && github.repository == 'gap-system/gap' }}
+    if: always() && github.event_name != 'pull_request' && github.repository == 'gap-system/gap'
```

Comment:
Wait, you can commit the `${{ ... }}$` ? On https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions it sounds as if that's required?! (Their docs really could be better)

Reply:
On https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif it says

> When you use expressions in an if conditional, you may omit the expression syntax (${{ }}) because GitHub automatically evaluates the if conditional as an expression, unless the expression contains any operators. If the expression contains any operators, the expression must be contained within ${{ }} to explicitly mark it for evaluation.

Since there are operators in there, I assume we *do* need it?

Reply:
Well, we don't _need_ it, because I've found that it does work without the `${{ ... }}`, but according to the documentation, it is needed. I can re-add the `${{ ... }}` if you insist.

We have been using this kind of syntax already for a while without problems; see several lines further down the workflow file:
```
if: steps.should_notify.outputs.should_send_message == 'yes'
```

Reply:
I am simply a bit wary about relying on undocumented behavior that explicitly contradicts the documentation, even if it works right now anyway...